Offline building should be clarified (plus clarifications for new builders re: gverify)

midnightmagic opened the pull request:
We should not imply they are to users who might want to torify or build offline.

Also, corrections to build process; including gverify examples for new builders.

laanwj commented:
As I've explained on IRC yesterday, offline builds *are* possible without manual patching.

- Download the dependency sources manually or use `make download` (as mentioned)
- Make sure that the git repositories with bitcoin (`https://github.com/bitcoin/bitcoin.git`) as well as the signatures (`https://github.com/bitcoin/bitcoin-detached-sigs.git`, only necessary for *signer.yml) are available locally, then to `gbuild` pass `--url bitcoin=${P1} --url signature=${P2}`. `P1` and `P2` must be set to the local paths of said repositories.

Anything missing?

midnightmagic commented:
Correcting the pullreq: I think it's valuable to provide at least some direction on doing an offline build. I suppose in the end there should also be either an offline apt-cacher-ng or one that's been torified but..

laanwj commented:
I'd suppose when you provide apt-cacher-bg with all packages, it doesn't need to access any from the internet anymore. But I haven't checked.

I suppose it would be possible to add socks support to apt-cacher-ng but torifying is most throughly and safely done by routing all of a user's network traffic through Tor using iptables.

midnightmagic commented:
After apt-cacher-ng is populated you can set it to be offline in its config file. Unfortunately it looks like the plain archive.ubuntu.com chokes up apt-cacher-ng as per:

https://bugs.launchpad.net/ubuntu/+source/apt-cacher/+bug/1358883

—of all the weird things, but switching it to us.archive.ubuntu.com corrects this.

Other than that, the --url bitcoin={},signature={} works well.

jgarzik commented:
ACK

laanwj commented on doc/release-process.md:
```diff
@@ -52,28 +67,44 @@ Release Process
+
 	./bin/gbuild --commit bitcoin=v${VERSION} ../bitcoin/contrib/gitian-descriptors/gitian-osx.yml
 	./bin/gsign --signer $SIGNER --release ${VERSION}-osx-unsigned --destination ../gitian.sigs/ ../bitcoin/contrib/gitian-descriptors/gitian-osx.yml
+	./bin/gverify -v -d ../gitian.sigs/ -r ${VERSION}-osx-unsigned /v/soft/btc/bitcoin/contrib/gitian-descriptors/gitian-osx.yml
```
path? `/v/soft/btc/bitcoin/contrib/gitian-descriptors/gitian-osx.yml`

midnightmagic commented:
Corrected path.

jgarzik commented:
needs rebase to fix conflicts; looks ready to merge.

midnightmagic committed:
Clarifying offline build process using gbuild --url and noting it is not
done automatically.

At some point along the line, fully offline builds were no longer happening
when strictly following the release-process.md instructions.

We should ensure that users who might want to torify or build offline need
to take extra steps to remain offline.

Also, corrections to build process: including gverify examples for new builders.

midnightmagic commented:
Fixed a couple more spelling errors, re-read it. Should be okay as long as nothing big has changed in the build. At the very least, it has some more detail to help new users to get into gitian building, which is good I think.
